app/api/entities/entitiesValidator.js

```diff
@@ -0,0 +1,9 @@
+import { validateEntity } from './entitySchema';
+
+const validator = {
+  save(entity) {
```

Comment:
I don't fully understand the purpose of this construct. If we need to provide a per-function validation helper here, why not just call validateEntity() in the entities.js functions?

Reply:
That's a good question. The rationale was to allow the developer to write and test the functionality of `entities.save()` without having to worry about validation, and to have that taken care of separately. This was an experiment and I can't say that it's any better than the alternative. I also tried this to allow me to add validation without the burden of updating every single unit test case, since many tests only use a subset of the data that is relevant to the behaviour being tested and this subset will often fail validation. But I will have to update tests either way, so this may not have been as thoughtful an idea.

Reply:
Just to add a clarification on the tests, I still need to update the tests of external modules which call `entities.save()` since external modules work with the "wrapped" version of `entities`, which performs validation.

Reply:
@joan @konzz @RafaPolit what is your view on this approach?

Reply:
I fixed the unit tests, most of them were due to mocking not working as expected with the wrapped object, I updated the implementation of `wrapValidation` to fix it. Maybe this a red flag, too?

Reply:
The extra bind is not a red flag in itself, but I do think the wrapValidation generally adds complexity and is not adding enough benefit to make up for that. So I would vote to remove it...

Reply:
I've removed `wrapValidation`.